BlueDB/src/main/java/org/bluedb/api/SegmentSize.java

```diff
@@ -0,0 +1,27 @@
+package org.bluedb.api;
+
+import org.bluedb.api.keys.BlueKey;
+import org.bluedb.api.keys.HashGroupedKey;
+import org.bluedb.api.keys.IntegerKey;
+import org.bluedb.api.keys.LongKey;
+import org.bluedb.api.keys.TimeKey;
+
+public class SegmentSize<T extends BlueKey> {
+
+	public static final SegmentSize<TimeKey> TIME_1_HOUR = new SegmentSize<>(3_600_000L);
+	public static final SegmentSize<TimeKey> TIME_1_DAY = new SegmentSize<>(24 * 3_600_000L);
+	public static final SegmentSize<LongKey> LONG_DEFAULT = new SegmentSize<>(64L);
+	public static final SegmentSize<IntegerKey> INT_DEFAULT = new SegmentSize<>(256L);
+	public static final SegmentSize<HashGroupedKey<?>> HASH_DEFAULT = new SegmentSize<>(524288L);

```

Comment:
I don't know that we should put default in the names. I don't know how much I like the current defaults for long, int, and hash. If we end up changing the default before end of June (I think we should consider it) then what do we call the new default? I think we should chat about what options to roll out with initially. We might want to call the current ones low density or something. And make high density options that end up being the default for new collections. Obviously legacy collections need to use the configuration that was originally the default.

Reply:
Actually, I changed my mind (see my comment on the `SegmentSizeSettings` class). I don't think the public user options need to contain specific information about their size. It should just represent what the user requested. It should be mapped to a `SegmentSizeSettings` option based on current defaults which gets persisted and contains all the information that the path manager needs. So having and option called INT_DEFAULT is fine I think. We should just have the ability to change what that means under the covers. Which may not end up using the same segment size as before.